scripts/normalize/lib/docs1-link-rewriter.js

```diff
@@ -0,0 +1,5204 @@
+  {
+    // first try just looking at previous sublings
+    const parent = ancestors[ancestors.length-1];
+    const index = parent.children.indexOf(node);
+    const textSiblings = parent.children.slice(0, index);
+    const result = [];
+    lookbackN = Math.min(lookbackN || textSiblings.length, textSiblings.length);
+    for (let i=textSiblings.length; lookbackN && i>=0; --i)
+    {
+      const segment = mdast2string({children: [textSiblings[i]]}).trim();
+      if (segment.length)
+      {
+        result.unshift(segment);
+        lookbackN--;
+      }
+    }
+    if (result.length) return result.join(' ');
+
+    // nothing there? Go up a level and do the same thing, but only look back one node
+    if (ancestors.length > 1)
+      return getContext(parent, ancestors.slice(0, -1), 1);
+    return "";
+  }
+}
+
+const titleMap = {
+  'EDB Advanced Server Installation Guide': '/epas/latest/',
+  'EDB Postgres Advanced Server Installation Guide': '/epas/latest/',
+  'appropriate version of the EDB Postgres Advanced Server Installation Guide': '/epas/latest/',
+  'EDB Postgres Advanced Server Guide': '/epas/latest/',
+  'Advanced Server guides': '/epas/latest/',
+  'complete library of Advanced Server documentation': '/epas/latest/',
+  'complete set of Advanced Server guide': '/epas/latest/',
+
+  'EDB Postgres Advanced Server OCL Connector Guide': '/ocl_connector/latest/',
+  'EDB Postgres Advanced Server ECPG Connector Guide': '/epas/latest/ecpgplus_guide/',
+  'Database Compatibility for Oracle Developer’s Catalog Views Guide': '/epas/latest/epas_compat_cat_views/',
+  'Database Compatibility for Oracle Developer’s Tools and Utilities Guide': '/epas/latest/epas_compat_tools_guide/',
+  'Database Compatibility for Oracle Developers Built-in Package Guide': '/epas/latest/epas_compat_bip_guide/',
+  'Database Compatibility for Oracle Developers Catalog Views Guide': '/epas/latest/epas_compat_cat_views/',
+  'Database Compatibility for Oracle Developers SQL Guide': '/epas/latest/epas_compat_sql/',
+  'Database Compatibility for Oracle Developer’s Guide': '/epas/latest/epas_compat_ora_dev_guide/',
+  'Database Compatibility for Oracle Developers Reference Guide': '/epas/latest/epas_compat_reference/',
+
+  'EDB Backup and Recovery User Guide': '/bart/latest/bart_user/',
+  'EDB Backup and Recovery Tool User Guide': '/bart/latest/bart_user/',
+  'EDB Backup and Recovery Us': '/bart/latest/bart_user/',
+  'EDB Backup and Recovery Reference Guide': '/bart/latest/bart_ref/',
+  'EDB Backup and Recovery Installation and Upgrade Guide': '/bart/latest/bart_inst/',
+  'BART Installation Guide': '/bart/latest/bart_inst/',
+  'BART Installation and Upgrade Guide': '/bart/latest/bart_inst/',
+  "PEM Administrator's Guide": '/pem/latest/pem_admin/',
+  'PEM Agent Guide': '/pem/latest/pem_agent/',
+  'PostgreSQL core documentation': 'https://www.postgresql.org/docs/',
+};
+
+
+const urlMap = {
+  /* these are unique to this file */
```

Comment:
This works, but I wouldn't be upset if you wanted to break this out into a `.json` file specific for this script ;)